operator/pkg/controllers/etcd/bootstrapconfig.go

```diff
@@ -0,0 +1,82 @@
+package etcd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/awslabs/kubernetes-iteration-toolkit/operator/pkg/apis/controlplane/v1alpha1"
+	"github.com/awslabs/kubernetes-iteration-toolkit/operator/pkg/utils/object"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Reconciles a ConfigMap that contains a bash script used for etcd cluster bootstrapping. This ConfigMap is mounted
+// as a volume in pods created as part of the etcd StatefulSet. This bash script is available in the container at
+// /etc/kubernetes/bootstrap.sh and executed in the container on start-up. Container arguments to this script
+// are passed directly to the etcd binary. Note that we manually construct a v1.ConfigMap struct rather than calling
+// object.GenerateConfigMap because double quote characters are not correctly populated in the ConfigMap by the UniversalDecoder.
+func (c *Controller) reconcileBootstrapConfig(ctx context.Context, controlPlane *v1alpha1.ControlPlane) error {
+	configMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      bootstrapConfigMapName(controlPlane.ClusterName()),
+			Namespace: controlPlane.Namespace,
+		},
+		Data: map[string]string{
+			"bootstrap.sh": bootstrapCmd(controlPlane),
+		},
+	}
+	return c.kubeClient.EnsurePatch(ctx, &v1.ConfigMap{}, object.WithOwner(controlPlane, configMap))
+}
+
+func bootstrapCmd(controlPlane *v1alpha1.ControlPlane) string {
+	etcdctlEndpoint := fmt.Sprintf("https://%s:2379", serviceFQDN(controlPlane))
+	return fmt.Sprintf(bootstrapScript, etcdctlEndpoint, bootstrapAdvertizePeerURL(controlPlane))
+}
+
+func bootstrapConfigMapName(clusterName string) string {
+	return fmt.Sprintf("%s-etcd-bootstrap", clusterName)
+}
+
+// bootstrapAdvertizePeerURL is the same as advertizePeerURL but uses curly brackets rather than parentheses around
+// the NODE_ID environment variable. When executing the bootstrap Configmap as a bash script in the etcd container,
+// we will get this error if we try using parentheses: ./etc/kubernetes/bootstrap.sh: line 27: NODE_ID: command not found
+func bootstrapAdvertizePeerURL(controlPlane *v1alpha1.ControlPlane) string {
```

Comment:
Does this need to be a function, can just use a variable like we are doing for `etcdctlEndpoint`?